### PR TITLE
Add constructor signature to FastifyError

### PR DESCRIPTION
**Pull request by initplatform**
Without the constructor signature, typescript will error when you try and use the results of createError

Screen Shot 2020-07-10 at 10 01 50 AM

### CODE REVIEW

**Comment by initplatform** [comment]
Actually, this doesn't work. Don't merge it. 

I'm getting this:

`TypeError [ERR_INVALID_ARG_TYPE]: The "superCtor.prototype" property must be of type object. Received undefined`

**Review by mcollina — APPROVED** [review]
lgtm

File: index.d.ts
Comment (Ethan-Arrowood) [inline index.d.ts:1]: this needs to be a class

**Review by Ethan-Arrowood — CHANGES_REQUESTED** [review]

File: index.d.ts
Comment (fox1t) [reply index.d.ts:1]: An interface with a new() should be the same as a class.

Comment (fox1t) [reply index.d.ts:1]: Ok, I double checked this: it should work only when also a var declared using the same name as interface that represents the actual instance. It is better and more simple to use class in this case. :)

Comment (fox1t) [inline index.d.ts:1]: Class should be used instead of interface

**Review by fox1t — CHANGES_REQUESTED** [review]

**Comment by SnirShechter** [comment]
Can anyone merge this?

**Comment by initplatform** [comment]
@hixus. Looks like I should close this in favor of https://github.com/fastify/fastify-error/pull/6 ?

**Comment by mcollina** [comment]
I'll leave this to @fastify/typescript

**Comment by fox1t** [comment]
I think we can move to https://github.com/fastify/fastify-error/pull/6 for this specific fix.

**Pull request closed without merging**
